src/python/pants/backend/jvm/tasks/jvm_compile/zinc/zinc_compile.py

```diff
@@ -426,6 +453,9 @@ def _compile_hermetic(self, jvm_options, ctx, classes_dir, jar_file,
       classpath_entry.directory_digest for classpath_entry in scalac_classpath_entries
     )
 
+    _, nailgun_client_snapshot = NailgunClientBinary.global_instance().hackily_snapshot(self.context)
```

Comment:
This feels like something which should be done in the hermetic execution stack, rather than in the consumer of hermetic execution.

Can you file a ticket and add a TODO to fix this up as a follow-up?

Reply:
Could probably tack it in on to #8480.